monai/transforms/spatial/array.py

```diff
@@ -2609,3 +2620,167 @@ def _get_params(
+        start_pos: Sequence[int] = (),
+        fix_num_patches: Optional[int] = None,
+        overlap: Union[Sequence[float], float] = 0.0,
+        sort_fn: Optional[Union[Callable, str]] = None,
+        pad_mode: Union[NumpyPadMode, PytorchPadMode, str] = NumpyPadMode.CONSTANT,
+        pad_opts: Optional[Dict] = None,
+    ):
+        self.patch_size = ensure_tuple(patch_size)
+        self.start_pos = ensure_tuple(start_pos)
+        self.pad_mode: NumpyPadMode = convert_pad_mode(dst=np.zeros(1), mode=pad_mode)
+        self.pad_opts = {} if pad_opts is None else pad_opts
+        self.overlap = overlap
+        self.fix_num_patches = fix_num_patches
+        self.sort_fn: Optional[Callable]
+        if isinstance(sort_fn, str):
+            if sort_fn == "random":
+                self.sort_fn = np.random.random
+            if sort_fn == "min":
+                self.sort_fn = self.get_patch_sum
+            if sort_fn == "max":
+                self.sort_fn = self._get_negative_patch_sum
+            else:
+                ValueError(f'sort_fn should be either "min", "max", or "random", "{sort_fn}" was given.')
+        else:
+            self.sort_fn = sort_fn
+
+    @staticmethod
+    def get_patch_sum(x):
+        return x[0].sum()
+
+    @staticmethod
+    def _get_negative_patch_sum(x):
+        return -x[0].sum()
+
+    def __call__(self, array: NdarrayOrTensor):
+        # create the patch iterator which sweeps the image row-by-row
+        array_np, *_ = convert_data_type(array, np.ndarray)
+        patch_iterator = iter_patch(
+            array_np,
+            patch_size=(None,) + self.patch_size,  # expand to have the channel dim
+            start_pos=(0,) + self.start_pos,  # expand to have the channel dim
+            overlap=self.overlap,
+            copy_back=False,
+            mode=self.pad_mode,
+            **self.pad_opts,
+        )
+        if self.sort_fn is not None:
+            output = sorted(patch_iterator, key=self.sort_fn)
+        else:
+            output = list(patch_iterator)
+        if self.fix_num_patches:
+            output = output[: self.fix_num_patches]
+            if len(output) < self.fix_num_patches:
+                if self.pad_opts.get("constant_values"):
+                    patch = np.ones((array.shape[0], *self.patch_size)) * self.pad_opts["constant_values"]
+                else:
+                    patch = np.zeros((array.shape[0], *self.patch_size))
+                location = np.zeros((3, len(self.patch_size)))
+                output += [(patch, location)] * (self.fix_num_patches - len(output))
+
+        output = [convert_to_dst_type(src=patch, dst=array)[0] for patch in output]
+
+        return output
+
+
+class RandGridPatch(GridPatch, RandomizableTransform):
+    """
+    Extract all the patches sweeping the entire image in a row-major sliding-window manner with possible overlaps,
+    and with random offset for the starting position of upper left corner of the image.
```

Comment:
By "upper left" corner you mean the minimal corner? To cover 2D and 3D volumes I suggest we refer to the minimal corner at index (0,0) or (0,0,0).